src/components/MyAccount/RequestsTab/FreezeButton.tsx

```diff
@@ -1,15 +1,47 @@
 import { useState } from "react"
 import type { Hold, Patron } from "../../../types/myAccountTypes"
-import { Button } from "@nypl/design-system-react-components"
+import {
+  Box,
+  Button,
+  Heading,
+  Icon,
+  useModal,
+  Text,
+} from "@nypl/design-system-react-components"
+import styles from "../../../../styles/components/MyAccount.module.scss"
 
 const FreezeButton = ({ hold, patron }: { hold: Hold; patron: Patron }) => {
   const [frozen, setFrozen] = useState(hold.frozen)
   const [isDisabled, setIsDisabled] = useState(false)
+  const { onOpen: openModal, onClose: closeModal, Modal } = useModal()
+  const modalProps = {
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Text sx={{ marginLeft: "l", marginRight: "m" }}>
+          Please try again.
+        </Text>
+      </Box>
+    ),
+    closeButtonLabel: "OK",
+    headingText: (
+      <Heading className={styles.modalHeading}>
+        <>
+          <Icon size="large" name="errorFilled" color="ui.error.primary" />
+          <Text sx={{ marginBottom: 0 }}> Freezing this hold failed </Text>
+        </>
+      </Heading>
+    ),
+    onClose: () => {
+      closeModal()
+      setFrozen((frozen) => !frozen)
+      setIsDisabled(false)
+    },
```

Comment:
This may need a bit more work. I tested this locally by removing the conditional mentioned in the description. Then I intentionally messed up the fetch request url in the `handleFreezeClick` function. Because it's wrong, the `response.status` is 404 (I removed "a" from `.../holds/update/...`).

When I click to freeze the hold, the modal popups correctly and shows the error message. But when I click "OK", the button text reads "Unfreeze". I think it's happening here where setFrozen is happening optimistically.

Reply:
Hmm am I missing some step in the logic here or is it just that when the **failure** modal closes, the frozen state of the button should never change? So this line should just be removed period

Reply:
yeah, I think it's basically just remove this line but test it in case there are scenarios I did not test (I only really tested the scenario I wrote)

Reply:
Ok- removed that line and made the test of this case more specific